java/flight/src/test/java/org/apache/arrow/flight/TestServerOptions.java

```diff
@@ -128,4 +132,20 @@ public void domainSocket() throws Exception {
       }
     }
   }
+
+  @Test
+  public void checkReflectionMetadata() {
+    // This metadata is needed for gRPC reflection to work.
+    final ExecutorService executorService = Executors.newSingleThreadExecutor();
+    try (final BufferAllocator allocator = new RootAllocator(Integer.MAX_VALUE)) {
+      final FlightBindingService service = new FlightBindingService(allocator, new NoOpFlightProducer(),
+          ServerAuthHandler.NO_OP, executorService);
+      final ServerServiceDefinition definition = service.bindService();
+      assertNotNull(definition.getServiceDescriptor().getSchemaDescriptor());
+      definition.getMethods().forEach(method -> assertNotNull(method.getMethodDescriptor().getSchemaDescriptor()));
+      definition.getServiceDescriptor().getMethods().forEach(method -> assertNotNull(method.getSchemaDescriptor()));
```

Comment:
Is it possible to do stronger validation here?

Reply:
I've reworked the test to both make sure that all methods are present, and that the reflection objects are equal between both places where they are accessible and equal to the original object.